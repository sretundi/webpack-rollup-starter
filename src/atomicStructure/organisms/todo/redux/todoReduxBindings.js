
import { bindActionCreators } from 'redux';

import todoActions from './actions.js';

export const mapStateToProps = state => ({
    todoList: state.todoList
});

export const mapDispatchToProps = (dispatch) => {
    return {
        actions: bindActionCreators(todoActions, dispatch)
    }
}
      
