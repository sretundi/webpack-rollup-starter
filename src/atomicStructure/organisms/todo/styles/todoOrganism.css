
.todoOrganism {
    width: 400px;
}

