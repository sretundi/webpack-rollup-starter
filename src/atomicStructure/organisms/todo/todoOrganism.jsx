import React from 'react';
import { connect } from 'react-redux';

import { mapStateToProps, mapDispatchToProps } from './redux/todoReduxBindings.js';

import styles from './styles/todoOrganism.css';

import HorizontalUserInput from '../../molecules/userInputs/horizontaluserInput.jsx';
import TodoList from '../../molecules/todoList/todoList.jsx'

class TodoOrganism extends React.Component{
    constructor(props) {
        super(props);
        this.state = {
            editableItem: {
                index: null,
                isEditing: false
            },
            inputText: '',
            editedText: '',
        }
    }

    onEditTodo(text, index) {
        this.setState((prevState, props) => {
            return {
                editableItem: {
                    index,
                    isEditing: true
                },
                editedText: text
            }
        })
    }

    inputOnChange(text) {
        this.setState({
            inputText: text
        })
    }
    
    submitTodo() {
        this.props.actions.addTodo(this.state.inputText);
        this.setState({
            inputText: '',
        })
    }

    editInputChange(text) {
        this.setState({
            editedText: text,
        })
    }

    onSubmitEdit() {
        this.setState(
            {
                editableItem: { 
                    index: this.state.editableItem.index,
                    isEditing: false 
                }
            }, 
            () => this.props.actions.editTodo(this.state.editedText, this.state.editableItem.index)
        )
    }

    render() {
        return (
            <div
                className={styles.todoOrganism}
            >
                <HorizontalUserInput 
                    onClick={this.submitTodo.bind(this)} 
                    onChange={this.inputOnChange.bind(this)}
                    value={this.state.inputText}
                    buttonLabel='Submit'
                />
                <TodoList
                    deleteButtonOnClick={this.props.actions.removeTodo}
                    editButtonOnClick={this.onEditTodo.bind(this)}
                    items={this.props.todoList} 
                    editableItem={this.state.editableItem}
                    onInputChange={this.editInputChange.bind(this)}
                    editedInput={this.state.editedText}
                    saveEditedChanges={this.onSubmitEdit.bind(this)}
                />

            </div>
        )
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(TodoOrganism);
export {TodoOrganism}
