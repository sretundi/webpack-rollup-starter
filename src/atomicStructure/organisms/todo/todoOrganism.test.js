
import React from 'react';
import Renderer from 'react-test-renderer';

import { TodoOrganism } from './todoOrganism.jsx';
import { mapDispatchToProps } from './redux/todoReduxBindings.js';

jest.mock('../../molecules/userInputs/horizontaluserInput.jsx', () => 'UserInputs');
jest.mock('../../molecules/todoList/todoList.jsx', () => 'todoList');

it('Todo organism renders correctly', () => {
    const tree = Renderer
        .create(
            <TodoOrganism 
                {...mapDispatchToProps()}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});