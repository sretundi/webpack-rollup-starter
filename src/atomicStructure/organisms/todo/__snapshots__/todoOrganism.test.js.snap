// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo organism renders correctly 1`] = `
<div
  className="todoOrganism"
>
  <UserInputs
    buttonLabel="Submit"
    onChange={[Function]}
    onClick={[Function]}
    value=""
  />
  <todoList
    deleteButtonOnClick={[Function]}
    editButtonOnClick={[Function]}
    editableItem={
      Object {
        "index": null,
        "isEditing": false,
      }
    }
    editedInput=""
    onInputChange={[Function]}
    saveEditedChanges={[Function]}
  />
</div>
`;
